reducer/first.go

```diff
@@ -18,8 +18,16 @@ func (fp *FirstProto) Instantiate() Interface {
 	return &First{Resolver: fp.resolver}
 }
 
-func NewFirstProto(target string, field expr.FieldExprResolver) *FirstProto {
-	return &FirstProto{target, field}
+func (fp *FirstProto) TargetResolver() expr.FieldExprResolver {
+	return fp.tresolver
+}
+
+func NewFirstProto(target string, tresolver, resolver expr.FieldExprResolver) *FirstProto {
+	return &FirstProto{
+		target:    target,
+		tresolver: tresolver,
+		resolver:  resolver,
+	}
```

Comment:
We usually put constructors immediately after the type definition, not after its methods.